docs/rfc-fluffy-unicorn.md

```diff
@@ -0,0 +1,152 @@
+# Nimona: Fluffy-Unicorn
+Fluffy-Unicorn is a self-describing, type-hinted, encoding-
+independent, language-agnostic, and consistently hash-able data format.
+(_It sounds fancier than it really is, I promise._)
+
+It does not define how data should be encoded or decoded when being transported. It can use any data format such as [JSON], [CBOR], [MsgPack], [Cap-n-proto], and others.
+
+Its purpose is to be able to define (relatively) strongly-typed and self-defined objects that can be transported through any means and encoding while still being able to produce the same cryptographic hash.
+
+## Overview
+__First things first:__ Fluffy-Unicorn is based on the work of a number of projects that have been adapted to work together and fit the requirements of nimona. [JSON], [JSON-LD], [JSON-Schema], [Tagged JSON], and [Object hash].
+
+## Types
+Fluffy-Unicorn has its own list of basic types and structures, more closely resembling the ones of [JSON], other than the split of number to int/uint/float, and the addition of a binary type.
+
+* Object - A collection of key/value pairs
+* Array - An ordered list of values
+* String
+* Integer
+* Unsigned Integer
+* Float
+* Data (binary data)
+* Boolean
+
+Continuing its comparison with JSON, there are some restrictions that are not present in the JSON format.
+
+* The top-level Fluffy-Unicorn-Object must always be an object
+* All values of an array must be of the same type.
+* A number of key names are reserved.
+
+## Type Hinting
+In order to allow Fluffy-Unicorn-Objects to be encoding and language agnostic, the types of all values have to be able to survive multiple re-encodings between incompatible formats, languages, and libraries.
+
+Since we can’t depend on each encoding format to keep the correct type, we provide a hint for the value’s type as part of the key. 
+
+```json
+{
+  "some-string:s": "bar",
+  "nested-object:o": {
+    "unsigned-number-one:u": 1,
+    "array-of-ints:a<i>": [-1, 0, 1]
+  }
+}
+```
+
+Hints for all of the supported types and structures:
+
+* `s` string
```

Comment:
I'd go for the longer names, but see you took inspiration from TJSON. Considering TJSON is not widely adopted, I'd dare say go for the more verbose and less obscure option?